python_modules/dagster/dagster/_core/definitions/multi_asset_sensor_definition.py

```diff
@@ -893,19 +892,26 @@ def get_asset_cursor_with_advances(
             latest_unconsumed_record_by_partition = (
                 initial_asset_cursor.trailing_unconsumed_partitioned_event_ids
             )
-            unconsumed_events = list(context.get_trailing_unconsumed_events(asset_key)) + list(
-                context.instance.get_event_records(
-                    EventRecordsFilter(
-                        event_type=DagsterEventType.ASSET_MATERIALIZATION,
+
+            materialization_events = []
+            has_more = True
+            cursor = None
+            while has_more:
+                result = context.instance.fetch_materializations(
+                    AssetRecordsFilter(
                         asset_key=asset_key,
-                        after_cursor=latest_consumed_event_id_at_tick_start,
-                        before_cursor=greatest_consumed_event_id_in_tick,
+                        after_storage_id=latest_consumed_event_id_at_tick_start,
+                        before_storage_id=greatest_consumed_event_id_in_tick,
                     ),
                     ascending=True,
+                    limit=FETCH_MATERIALIZATION_BATCH_SIZE,
+                    cursor=cursor,
                 )
-                if greatest_consumed_event_id_in_tick
-                > (latest_consumed_event_id_at_tick_start or 0)
-                else []
```

Comment:
was this short-circuit check to not make the query needed in the first place not useful?